python_ta/checkers/global_variables_checker.py

```diff
@@ -76,6 +76,8 @@ def _get_child_disallowed_global_var_nodes(node):
     global, non-constant variable. 
     """
     node_list = []
+    if isinstance(node.parent, astroid.ListComp) or isinstance(node.parent, astroid.Comprehension):
```

Comment:
Thanks, I've added the checks for additional nodes.
Nice point about the nesting, although I can't think of a case where this doesn't work on nested expressions, so maybe the current approach is acceptable?